cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -193,9 +194,8 @@ public void shutdown() {
   /**
    * Get the state of the load monitor.
    */
-  public LoadMonitorState state(OperationProgress operationProgress) {
```

Comment:
What's the reason for adding the clusterAndGeneration parameter for this method and the meetCompletenessRequirements method?

Reply:
This method is called from `KafkaCruiseControl#state(OperationProgress)`, which also calls `GoalOptimizer#state(MetadataClient.ClusterAndGeneration)` in the same method. Previously, both `GoalOptimizer#state` and `LoadMonitor#state` methods used to call `_metadataClient.refreshMetadata()` independently. This would potentially mean that each method would poll a broker for metadata update. In particular, `GoalOptimizer#state` would even poll a broker for metadata update for each goal.

This change ensures that rather than having `1 + #numberOfGoals` to a broker, we only have a single call for metadata update.

Reply:
Talked offline with Efe, and this change will prevent the corner case of refreshing the metadata twice during the metadata TTL expirations.